### PR TITLE
Consolidate TFM signing

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Use the same signing code for every board.

Depends on #31913

### CODE REVIEW

File: boards/CMakeLists.txt [hunk 1]
```diff
@@ -12,3 +12,125 @@ if(EXISTS ${BOARD_DIR}/CMakeLists.txt)
...
+        --layout ${PREPROCESSED_FILE_NS}
+        -k ${CONFIG_TFM_KEY_FILE_NS}
+        --public-key-format ${TFM_PUBLIC_KEY_FORMAT}
+        --align 1
+        -v ${TFM_IMAGE_VERSION_NS}
+        --pad
+        --pad-header
+        ${ADD_NS_IMAGE_MIN_VER}
+        -s auto
+        -H 0x400
```
Comment (hakonfam) [inline boards/CMakeLists.txt:70]: Size of header? Should we have a comment for these arguments?

**Review by hakonfam — APPROVED** [review]
Nice!

**Review by ioannisg — COMMENTED** [review]
Is there a way for the common cmake stuff for signing to reside in the TF-M CMakeLists.txt (currently in the module directory, but will move into the main tree). In this way we don't pollute the main boards' CMakeLists.txt file.

That's a question for @tejlmand

File: modules/Kconfig.tfm [hunk 2]
```diff
@@ -24,6 +24,7 @@ menuconfig BUILD_WITH_TFM
 	depends on TRUSTED_EXECUTION_NONSECURE
 	depends on TFM_BOARD != ""
 	depends on ARM_TRUSTZONE_M
+	imply BUILD_OUTPUT_HEX # For signing.
```
Comment (ioannisg) [inline modules/Kconfig.tfm:27]: No need for the inline comment, IMHO

**Comment by tejlmand** [comment]
> Is there a way for the common cmake stuff for signing to reside in the TF-M CMakeLists.txt (currently in the module directory, but will move into the main tree). In this way we don't pollute the main boards' CMakeLists.txt file.

I had a quick look, and initial feeling is that this should indeed be doable.
Haven't noticed anything that makes me think it should not be possible, but of course we need to ensure the post build commands are still executed in correct order, for those that are depending on other part.

As side note, I've noticed increased use of the extra post build command property and to ensure correct dependency handling, some of those should probably be changed into custom commands for proper dependency chain.

File: boards/CMakeLists.txt (hunk 1 above)
Comment (oyvindronningstad) [reply boards/CMakeLists.txt]: Used Kconfig var instead.

File: modules/Kconfig.tfm (hunk 2 above)
Comment (oyvindronningstad) [reply modules/Kconfig.tfm]: Done

File: modules/trusted-firmware-m/Kconfig [hunk 3]
```diff
@@ -25,6 +25,7 @@ menuconfig BUILD_WITH_TFM
 	depends on TRUSTED_EXECUTION_NONSECURE
 	depends on TFM_BOARD != ""
 	depends on ARM_TRUSTZONE_M
+	imply BUILD_OUTPUT_HEX
```
Comment (tejlmand) [inline modules/trusted-firmware-m/Kconfig:28]: is `imply` strong enough here ?
If there is not hex output, will the following build / signing / merge stages work as expected ?

Comment (oyvindronningstad) [reply modules/trusted-firmware-m/Kconfig:28]: No, it will fail with "file not found" or something.

Comment (oyvindronningstad) [reply modules/trusted-firmware-m/Kconfig:28]: Is `select` better?

Comment (tejlmand) [reply modules/trusted-firmware-m/Kconfig:28]: in that case, yes.

Comment (oyvindronningstad) [reply modules/trusted-firmware-m/Kconfig]: done

**Review by tejlmand — APPROVED** [review]

**Review by ioannisg — CHANGES_REQUESTED** [review]
Looks ok to me, but we probably need to fix the QEMU running for the integration samples, which seems to me it's currently broken.
I think something might be wrong with the executable paths @oyvindronningstad

**Review by ioannisg — COMMENTED** [review]
I think this needs some extra work @oyvindronningstad , particularly, for the QEMU target, which seems to be using the tfm_qemu.bin  and .hex respectively.

This is something we can clean up in the samples, but we also need to fix the consolidation, to work for the mps2 target, because even if i switch samples to use tfm_merged.hex, the qemu does not work.

**Comment by oyvindronningstad** [comment]
Ok, so just copying tfm_merged.hex to qemu.hex will fix it? We can do that of course. Is there a way to check whether qemu is available and only create it for those targets?

**Comment by ioannisg** [comment]
@oyvindronningstad i pushed a commit that does it (hopefully). Had not seen your comment , sorry.

**Comment by oyvindronningstad** [comment]
@ioannisg That looks even better, thanks!

**Review by ioannisg — APPROVED** [review]

**Pull request merged**